src/components/MovieCard/MovieCard.js

```diff
@@ -1,32 +1,81 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styles from './MovieCard.module.scss';
+import MovieOptionsPopup from '../MovieOptionsPopup/MovieOptionsPopup';
 
-const MovieCard = ({ isBuggy, title, year, posterUri }) => {
-  if (isBuggy) {
-    throw new Error('random error');
+class MovieCard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showOptionsPopup: false,
+    };
+
+    this.toggleOptionsPopup = this.toggleOptionsPopup.bind(this);
+    this.onOptionsPopupAction = this.onOptionsPopupAction.bind(this);
+    this.hideOptionsPopup = this.hideOptionsPopup.bind(this);
   }
 
-  return (
-    <div className={styles.MovieCard}>
-      <img src={posterUri} />
-      <div className={styles.descriptionContainer}>
-        <p className={styles.title}>{title}</p>
-        <p className={styles.year}>{year}</p>
-      </div>
-    </div>
-  );
-};
+  onOptionsPopupAction(action) {
+    const { optionsHandler, movie: { id } } = this.props;
+    optionsHandler(action, id);
+    this.hideOptionsPopup();
+  }
+
+  hideOptionsPopup() {
+    this.setState({ showOptionsPopup: false });
+  }
+
+  toggleOptionsPopup() {
+    const { showOptionsPopup } = this.state;
+    this.setState({ showOptionsPopup: !showOptionsPopup });
+  }
+
+  render() {
+    const { movie } = this.props;
```

Comment:
```js
const { title, year, posterUri } = this.props.movie;
```

Reply:
applied nested object destructuring:
`const { movie: { title, year, posterUri } } = this.props;`